File: Manuals/DCx_Camera_Interfaces_2018_09/DCx_Camera_SDK/Develop/Source/uc480_DotNet_VB_SimpleLive/Form1.vb

Public Class uEye_DotNet_Simple_Live

    Dim WithEvents camera As New uc480.Camera()
    Dim DisplayHandle As IntPtr
    Dim bLive As Boolean


    Private Sub uEye_DotNet_Simple_Live_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        DisplayHandle = DisplayWindow.Handle
        CameraInit()

    End Sub


    Private Sub EventHandler() Handles camera.EventFrame

        'Event
        Dim s32MemID As Int32
        camera.Memory.GetActive(s32MemID)
        camera.Display.Render(s32MemID, DisplayHandle, uc480.Defines.DisplayRenderMode.FitToWindow)

    End Sub


    Private Sub CameraInit()

        Dim s32MemID As Int32
        Dim statusRet As uc480.Defines.Status

        'Open Camera
        statusRet = camera.Init()
        If (statusRet <> uc480.Defines.Status.Success) Then

            MessageBox.Show("Camera initializing failed")

        End If

        'Allocate Memory
        statusRet = camera.Memory.Allocate(s32MemID, True)
        If (statusRet <> uc480.Defines.Status.Success) Then

            MessageBox.Show("Allocate Memory failed")

        End If

        'Start Live
        statusRet = camera.Acquisition.Capture()
        If (statusRet <> uc480.Defines.Status.Success) Then

            MessageBox.Show("Start Live Video failed")

        Else

            bLive = True

        End If

        CB_Auto_Gain_Balance.Enabled = camera.AutoFeatures.Software.Gain.Supported
        CB_Auto_White_Balance.Enabled = camera.AutoFeatures.Software.WhiteBalance.Supported

    End Sub


    Private Sub Button_Live_Video_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button_Live_Video.Click

        'Start Live Video
        If (camera.Acquisition.Capture() = uc480.Defines.Status.Success) Then

            bLive = True

        End If

    End Sub


    Private Sub Button_Stop_Video_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button_Stop_Video.Click

        'Stop Live Video
        If (camera.Acquisition.Stop() = uc480.Defines.Status.Success) Then

            bLive = False

        End If

    End Sub


    Private Sub Button_Freeze_Video_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button_Freeze_Video.Click

        'Freeze Video
        If (camera.Acquisition.Freeze() = uc480.Defines.Status.Success) Then

            bLive = False

        End If

    End Sub


    Private Sub Button_Load_Parameter_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button_Load_Parameter.Click

        'Load Parameters
        camera.Acquisition.Stop()
        camera.Parameter.Load("")

        If (bLive = True) Then

            If (camera.Acquisition.Capture() = uc480.Defines.Status.Success) Then

                bLive = True

            End If

        End If

    End Sub


    Private Sub CB_Auto_White_Balance_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CB_Auto_White_Balance.CheckedChanged

        'CheckBox Auto White Balance
        camera.AutoFeatures.Software.WhiteBalance.SetEnable(CB_Auto_White_Balance.Checked)

    End Sub


    Private Sub CB_Auto_Gain_Balance_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles CB_Auto_Gain_Balance.CheckedChanged

        'CheckBox Auto Gain Balance
        camera.AutoFeatures.Software.Gain.SetEnable(CB_Auto_Gain_Balance.Checked)

    End Sub


    Private Sub Button_Exit_Prog_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button_Exit_Prog.Click

        'Close the Camera
        camera.Exit()

        'Program Exit
        Close()

    End Sub

End Class

File: Manuals/DCx_Camera_Interfaces_2018_09/DCx_Camera_SDK/Develop/Source/uc480_DotNet_C#_SimpleLive_2/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SimpleLive
{
    public partial class uc480_DotNet_Simple_Live : Form
    {
        private uc480.Camera Camera;
        IntPtr displayHandle = IntPtr.Zero;
        private bool bLive          = false;

        public uc480_DotNet_Simple_Live()
        {
            InitializeComponent();

            displayHandle = DisplayWindow.Handle;
            InitCamera();
        }

        private void InitCamera()
        {
            Camera = new uc480.Camera();//Use only the empty constructor, the one with cameraID has a bug

            uc480.Defines.Status statusRet = 0;

            // Open Camera
            statusRet = Camera.Init();//You can specify a particular cameraId here if you want to open a specific camera

            if (statusRet != uc480.Defines.Status.SUCCESS)
            {
                MessageBox.Show("Camera initializing failed");
            }

            // Allocate Memory
            Int32 s32MemID;
            statusRet = Camera.Memory.Allocate(out s32MemID, true);
            if (statusRet != uc480.Defines.Status.SUCCESS)
            {
                MessageBox.Show("Allocate Memory failed");
            }

            // Start Live Video
            statusRet = Camera.Acquisition.Capture();
            if (statusRet != uc480.Defines.Status.SUCCESS)
            {
                MessageBox.Show("Start Live Video failed");
            }
            else
            {
                bLive = true;
            }

            // Connect Event
            Camera.EventFrame += onFrameEvent;

            CB_Auto_Gain_Balance.Enabled = Camera.AutoFeatures.Software.Gain.Supported;
            CB_Auto_White_Balance.Enabled = Camera.AutoFeatures.Software.WhiteBalance.Supported;
        }

        private void onFrameEvent(object sender, EventArgs e)
        {
            uc480.Camera Camera = sender as uc480.Camera;

            Int32 s32MemID;
            Camera.Memory.GetActive(out s32MemID);

            Camera.Display.Render(s32MemID, displayHandle, uc480.Defines.DisplayRenderMode.FitToWindow);
        }
        
        private void Button_Live_Video_Click(object sender, EventArgs e)
        {
            // Open Camera and Start Live Video
            if (Camera.Acquisition.Capture() == uc480.Defines.Status.SUCCESS)
            {
                bLive = true;
            }
        }

        private void Button_Stop_Video_Click(object sender, EventArgs e)
        {
            // Stop Live Video
            if (Camera.Acquisition.Stop() == uc480.Defines.Status.SUCCESS)
            {
                bLive = false;
            }
        }
        
        private void Button_Freeze_Video_Click(object sender, EventArgs e)
        {
            if (Camera.Acquisition.Freeze() == uc480.Defines.Status.SUCCESS)
            {
                bLive = false;
            }
        }

        private void CB_Auto_White_Balance_CheckedChanged(object sender, EventArgs e)
        {
            Camera.AutoFeatures.Software.WhiteBalance.SetEnable(CB_Auto_White_Balance.Checked);
        }

        private void CB_Auto_Gain_Balance_CheckedChanged(object sender, EventArgs e)
        {
            if (CB_Auto_Gain_Balance.Checked)
            {
                Camera.AutoFeatures.Software.Gain.SetEnable(true);
            }
            else
            {
                Camera.AutoFeatures.Software.Gain.SetEnable(false);
            }
        }

        private void Button_Load_Parameter_Click(object sender, EventArgs e)
        {
            Camera.Acquisition.Stop();

            Camera.Parameter.Load("");

            if (bLive == true)
            {
                Camera.Acquisition.Capture();
            }
        }

        private void Button_Exit_Prog_Click(object sender, EventArgs e)
        {
            // Close the Camera
            Camera.Exit();
            Close();
        }
    }
}
